yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -94,6 +98,14 @@ private[yarn] class YarnAllocator(
   // Additional memory overhead.
   protected val memoryOverhead: Int = sparkConf.getInt("spark.yarn.executor.memoryOverhead",
     math.max((MEMORY_OVERHEAD_FACTOR * executorMemory).toInt, MEMORY_OVERHEAD_MIN))
+
+  // Make the maximum executor failure check to be relative with respect to duration
+  private val relativeMaxExecutorFailureCheck = 
```

Comment:
Isn't this more easily solved by considering max failures to be per batch interval or something? why define another different window of time in another property?